registry/repository.go

```diff
@@ -59,6 +58,11 @@ type Repository interface {
 	// - https://docs.docker.com/registry/spec/api/#tags
 	// See also `Tags()` in this package.
 	Tags(ctx context.Context, fn func(tags []string) error) error
+
+	// Referrers returns the manifest descriptors directly referencing the given
+	// manifest descriptor.
+	// Reference: https://github.com/oras-project/artifacts-spec/blob/main/manifest-referrers-api.md
+	Referrers(ctx context.Context, desc ocispec.Descriptor, fn func(referrers []artifactspec.Descriptor) error) error
```

Comment:
We probably don't want to add `Referrers` here because not all registries support `Referrers` API.

Reply:
Moved this out.